### PR TITLE
chore: bump dependencies to their latest SemVer-compliant versions

### PR DESCRIPTION
**Pull request by ericcarraway**
#### DESCRIPTION:

The goal of this PR is to bump (dev)dependencies to their latest SemVer-compliant major versions, per the caret (`^`) annotations.  These are the versions of the (dev)dependencies that would be installed if a user ran `npm install` today.

This PR does not attempt to make any "major" version bumps.  It only makes "minor" and "patch" bumps.

[@types/jest](https://www.npmjs.com/package/@types/jest): `^25.2.1 -> ^25.2.3`
[@types/lodash](https://www.npmjs.com/package/@types/lodash): `^4.14.150 -> ^4.14.168`
[@typescript-eslint/eslint-plugin](https://www.npmjs.com/package/@typescript-eslint/eslint-plugin): `^2.31.0 -> ^2.34.0`
[@typescript-eslint/parser](https://www.npmjs.com/package/@typescript-eslint/parser): `^2.31.0 -> ^2.34.0`
[eslint-config-prettier](https://www.npmjs.com/package/eslint-config-prettier): `^6.11.0 -> ^6.15.0`
[eslint-plugin-prettier](https://www.npmjs.com/package/eslint-plugin-prettier): `^3.1.3 -> ^3.3.1`
[husky](https://www.npmjs.com/package/husky): `^4.2.5 -> ^4.3.8`
[jest](https://www.npmjs.com/package/jest): `^26.0.1 -> ^26.6.3`
[lint-staged](https://www.npmjs.com/package/lint-staged): `^10.2.2 -> ^10.5.4`
[prettier](https://www.npmjs.com/package/prettier): `^2.0.5 -> ^2.2.1`
[ts-jest](https://www.npmjs.com/package/ts-jest): `^25.5.0 -> ^25.5.1`
[typescript](https://www.npmjs.com/package/typescript): `^3.8.3 -> ^3.9.9`
[lodash](https://www.npmjs.com/package/lodash): `^4.17.15 -> ^4.17.21`

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,46 +1,52 @@
 {
-  "name": "@highcoast/convertkit-node",
+  "name": "convertkit-node",
```
Comment (ericcarraway) [inline package-lock.json:2]: `npm install` made this update.

`converkit-node` is the name listed in the `package.json`.  Perhaps this package was previously scoped to `@highcoast/convertkit-node`?

**Review by jrbaudin — DISMISSED** [review]

File: package-lock.json (hunk 1 above)
Comment (jrbaudin) [reply package-lock.json:2]: Yes, I believe so 🤔 :)

**Comment by jrbaudin** [comment]
Looks great @ericcarraway 🙌🏼

**Comment by ericcarraway** [comment]
Rebased from de0786cbdd833c6cbb4c21579ee56df29f29279e to c3f8f4036e67c7aa086aaadfe42131d19857f881 so that this PR is on top of the latest main branch.

**Review by jrbaudin — APPROVED** [review]

**Pull request merged**